js/nzcvm_config_map.js

```diff
@@ -51,17 +55,163 @@ let lastPos = null;
+    }
+
+    console.log(`Parsed ${locations.length} locations from file`);
+    return locations;
+}
+
+// Function to display location markers on the map with optimizations for large datasets
+function displayLocationMarkers(locations) {
+    // Don't proceed if no locations
+    console.log(locations)
+    if (locations.length === 0) return;
+
+    // Create a standard layer group instead of marker cluster
+    locationMarkersLayer = L.layerGroup().addTo(map);
+
+    for (const loc of locations) {
+        // Use circleMarker instead of marker for better performance with large datasets
+        const marker = L.circleMarker([loc.lat, loc.lng], {
+            radius: 4,
+            fillColor: '#3388ff',
+            color: '#fff',
+            weight: 1,
+            opacity: 1,
+            fillOpacity: 0.8,
+            title: loc.label
+        });
+
+        // Add a popup with the name
+        marker.bindPopup(loc.label);
+
+        // Store reference to the original data
+        marker.locationData = loc;
+
+        // Add marker to the collection and track it
+        locationMarkersLayer.addLayer(marker);
+        locationMarkers.push(marker);
+    }
+    // Create bounds that include both markers and rectangle
+    const group = new L.featureGroup([locationMarkersLayer, rectangle]);
+    console.log(group);
+}
+
+// Function to clear all location markers
+function clearLocationMarkers(resetInput = true) { // Parameter controls if file input value is cleared
+    if (locationMarkersLayer) {
+        map.removeLayer(locationMarkersLayer);
+        locationMarkersLayer = null;
+    }
+    locationMarkers = [];
+
+    // Reset the file input element value only if resetInput is true
+    if (resetInput) {
+        const fileInput = document.getElementById('location-file-input');
+        if (fileInput) {
+            fileInput.value = ''; // Clear the selected file in the input
+        }
+        // No need to reset any span text content
+    }
+}
+
+// Add event listener for file upload UI elements when the DOM is loaded
+document.addEventListener('DOMContentLoaded', function () {
+    const locationControls = createLocationUploadControls();
+    document.getElementById('map-container').appendChild(locationControls);
+
+    // Set up the event listener for file input
+    document.getElementById('location-file-input').addEventListener('change', handleLocationFileUpload);
+});
+
+// Create UI controls for location upload
+function createLocationUploadControls() {
+    const controlPanel = document.createElement('div');
+    controlPanel.className = 'location-upload-panel';
+    controlPanel.style.position = 'absolute';
+    controlPanel.style.top = '10px';
+    controlPanel.style.right = '10px';
+    controlPanel.style.zIndex = '1000';
+    controlPanel.style.backgroundColor = 'white';
+    controlPanel.style.padding = '10px';
+    controlPanel.style.borderRadius = '4px';
+    controlPanel.style.boxShadow = '0 0 10px rgba(0,0,0,0.1)';
```

Comment:
I'd rather see this extracted into the `style.css` file. See the changes to that file. 

```suggestion
    controlPanel.id = 'location-upload-panel';
```